src/hotspot/os/linux/os_linux.cpp

```diff
@@ -2174,7 +2174,12 @@ bool os::Linux::print_container_info(outputStream* st) {
   int i = OSContainer::active_processor_count();
   st->print("active_processor_count: ");
   if (i > 0) {
-    st->print_cr("%d", i);
+    if (ActiveProcessorCount > 0) {
+      st->print_cr("%d, but overwritten by flag ActiveProcessorCount: %d",
```

Comment:
This will print the same number (processor count) twice. I'd suggest to only add a blurb like `" overridden by -XX:ActiveProcessorCount"` in case the flag is set and not print the blurb otherwise.

Reply:
Asserting the new output in the jtreg test would be good to have too.

Reply:
This format is fine me with me. You want to see both the container processor count and the actual count being used.

Reply:
Hi  Severin , I adjusted the output following your suggestion  (David, in case ActiveProcessorCount  is set on the command line it is printed as well , see Arguments::print_summary_on , probably that's why Severin does not want it  here.
Regarding ,  whether other info we present can be overridden by flags, I think not in the container area  (not sure about the whole hs_err output).

Reply:
Sorry, I was wrong. I misread `int i = OSContainer::active_processor_count();` for `int i = os::active_processor_count();` which would then return the same value. As @dholmes-ora said, the original format was fine. Please add this in the test, though. Could be akin to `test/hotspot/jtreg/containers/docker/TestMisc.testPrintContainerInfo()`. I.e. set some CPU quota and also ActiveProcessorCount and assert the message in log output.

Reply:
> Sorry, I was wrong. I misread `int i = OSContainer::active_processor_count();` for `int i = os::active_processor_count();` which would then return the same value. As @dholmes-ora said, the original format was fine. Please add this in the test, though. Could be akin to `test/hotspot/jtreg/containers/docker/TestMisc.testPrintContainerInfo()`. I.e. set some CPU quota and also ActiveProcessorCount and assert the message in log output.

Hi Severin, I adjusted the output again; also adjusted the jtreg test containers/docker/TestMisc .

Best regards, Matthias